### PR TITLE
fix: support OTP2 scooter responses

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
OTP2 changes how scooters are represented. This PR aims to fully support OTP2 scooter responses

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]
LGTM

File: packages/core-utils/src/map.ts [hunk 1]
```diff
@@ -139,13 +145,18 @@ export function itineraryToTransitive(
       ) {
         // create a special place ID for car legs preceded by walking legs
         fromPlaceId = `itin_car_${streetEdgeId}_from`;
-      } else {
+      } else if (!fromPlaceId) {
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/map.ts:148]: to my understanding, without the `if` the clause is effected directly by the `leg.mode === "CAR"` `if` above it

**Comment by miles-grant-ibigroup** [comment]
I'll assign to @binh-dam-ibigroup once I fix the merge conflict

File: packages/core-utils/src/map.ts [hunk 2]
```diff
@@ -139,13 +145,18 @@ export function itineraryToTransitive(
...
         fromPlaceId = `itin_street_${streetEdgeId}_from`;
       }
 
       let toPlaceId;
       if (leg.to.bikeShareId) {
         toPlaceId = `bicycle_rent_station_${leg.to.bikeShareId}`;
+        // OTP2 scooter case
+        // Need to check next leg since this is a "to" place "
+        if (leg.mode === "SCOOTER" || itin.legs?.[idx + 1].mode === "SCOOTER") {
+          toPlaceId = `escooter_rent_station_${leg.to.bikeShareId}`;
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/map.ts:158]: In OTP2 they are the same which is why this went unnoticed! But consistency is good, so I've used `bikeShareId` in the from id

File: packages/core-utils/src/map.ts [hunk 3]
```diff
@@ -129,7 +129,13 @@ export function itineraryToTransitive(
...
       if (leg.from.bikeShareId) {
         fromPlaceId = `bicycle_rent_station_${leg.from.bikeShareId}`;
-        // TODO: does this need to change to be OTP2 compatible?
+        if (
+          // OTP2 Scooter case
+          leg.mode === "SCOOTER"
+        ) {
+          fromPlaceId = `escooter_rent_station_${leg.from.name}`;
+        }
+        // OTP1 Scooter case
```
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/map.ts]: Moved in 2b6cb62c

File: packages/core-utils/src/map.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (miles-grant-ibigroup) [reply packages/core-utils/src/map.ts:158]: 2b6cb62c

File: packages/itinerary-body/i18n/en-US.yml [hunk 4]
```diff
@@ -79,6 +79,7 @@ otpUi:
       BICYCLE_RENT {Bikeshare}
       CAR {{isCarHail, select, true {Ride} other {Drive}}}
       MICROMOBILITY {Ride}
+      SCOOTER {Ride}
```
Comment (binh-dam-ibigroup) [inline packages/itinerary-body/i18n/en-US.yml:82]: I think there should be a story involving an OTP2 scooter rental.

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Could you add a story to ItineraryBody with an OTP2 scooter rental, please?

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**